src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -647,6 +645,14 @@ private StringSubstringPrimitiveNode getSubstringNode() {
             return substringNode;
         }
 
+        private int executeNum2IntSnippetNode(VirtualFrame frame, Object... args) {
+            if (snippetNode == null) {
+                CompilerDirectives.transferToInterpreterAndInvalidate();
+                snippetNode = insert(new SnippetNode());
+            }
+
+            return (int) snippetNode.execute(frame, "Truffle::Type.rb_num2int(v)", args);
```

Comment:
Minor: SnippetNode asserts that the expression and the argument names never change, the name `"v"` could be encapsulated in the helper method as well. It has no meaning outside of the method.

Reply:
Yes, I saw it as well. But I'll change this to a call node very soon anyway.

Reply:
I fixed that and renamed the function to `toInt` as suggested by @eregon

Reply:
Thanks @wirthi